lib/entity_manager.js

```diff
@@ -55,9 +60,12 @@ EntityManager.prototype.query = function() {
   }
 
   var result = []
-  for (var id = 0; id < this._entityCounter; id++) {
-    if (this._entityFlag[id] === ENTITY_ALIVE && (this._entityMask[id] & mask) === mask) {
-      result.push(this._entityInst[id])
+
+  var entityMasks = Object.keys(this._entityCache)
```

Comment:
Sure! But I won't merge it immediately. I want to add more tests and benchmarks.